src/protos/query-item.proto

```diff
@@ -0,0 +1,11 @@
+syntax="proto3";
+
+package Indexer;
+
+import "protos/request-params.proto";
+import "protos/response.proto";
+
+service QueryItem {
+    rpc bangumiMoe(RequestParams) returns (BangumiMoeResponse) {};
```

Comment:
Can we have one API for both type?

Reply:
I cannot find a way to implement one API for gRPC, although protobuf support extends message. but it doesn't help here. You need to create two message for the different type anyway because the generic type is not supported.